### PR TITLE
Fixes #897 added context to application docs and moved some docs from…

### LINKED ISSUE #897: Add high level application lifecycle overview to docs
I think application documentation needs a high level overview that helps users understand the lifecycle of an application and how the commands and config all fit together.

Also while working on this, I think it would make sense to move the application specific docs in the install docs to the application docs.

Below is an outline of the docs I am thinking about.

## Application Lifecycle

 * **Initialization** : During initialization of an application, the following is done.
   * **Copy jars to DFS** :  TODO document props that influence this and why the jars are copied to DFS.
   * **Create Accumulo table** : TODO document configuring jars, props that influence this, that Fluo has server side code.
   * **Create Zookeeper node** :  TODO document that app props are stored here along with runtime info.
 * Start service processes :
   * **Oracle**: document what it does and that it must run for clients to work
   * **Worker** : document what it does.. Document that it uses jars from DFS and config from zookeeper for things like #worker threads
 * **External client**  .. TODO uses ORacle service process

### PR DESCRIPTION
**Pull request by keith-turner**
… install

### CODE REVIEW

**Comment by ctubbsii** [comment]
Looks good to me, but I'd shorten the log message so it fits in the subject line, and move the details to the body of the log message.

File: docs/applications.md [hunk 1]
```diff
@@ -17,8 +17,18 @@ limitations under the License.
...
+Once you have Fluo installed and running on your cluster, you can run Fluo applications consisting
+of clients and observers.  A client is a user started process that uses the Fluo API.  An Observer
+is user provided code run by Fluo Workers in response to notifications.  Notifications are set by
+Fluo transactions executing in a client or observer.  Clients and observers rely on a running Fluo
+oracle service.  This documentations will shows how to :
+
+ * Create a Fluo client
+ * Create a Fluo observer
+ * Initialize a Fluo Application
+ * Start and stop Fluo Oracle and Worker processes
```
Comment (mikewalch) [inline docs/applications.md]: Could be `Start and stop a Fluo application (which consists of oracle and worker processes)`

File: docs/applications.md [hunk 2]
```diff
@@ -70,31 +80,6 @@ FluoClient client = FluoFactory.newClient(config)
...
-
-  public static void main(String[] args) throws Exception {
-    try(FluoClient fluoClient = FluoFactory.newClient(fluoConfig)) {
-      //do stuff with Fluo
-    }
-  }
-}
-```
-
 ## Creating a Fluo observer
```
Comment (mikewalch) [inline docs/applications.md]: Now that this document has changed.  This section should be updated too.

File: docs/applications.md [hunk 3]
```diff
@@ -70,31 +80,6 @@ FluoClient client = FluoFactory.newClient(config)
...
 
 To create an observer, follow these steps:
```
Comment (mikewalch) [inline docs/applications.md]: Could be `To create an observer, follow these steps (which are described in more detail in the sections below):`

File: docs/applications.md [hunk 4]
```diff
@@ -141,12 +126,123 @@ To create an observer, follow these steps:
 3.  Build a jar containing these classes and include this jar in the `lib/` directory of your Fluo
     application.
 4.  Configure your Fluo instance to use this observer provider by modifying the Application section of
```
Comment (mikewalch) [inline docs/applications.md]: Should use `application` instead of `instance`

File: docs/applications.md [hunk 5]
```diff
@@ -141,12 +126,123 @@ To create an observer, follow these steps:
...
-    [fluo-app.properties].
+    [fluo-app.properties]. Set `fluo.observer.provider` to the observer provider class name.
 5.  Initialize Fluo.  During initialization Fluo will obtain the observed columns from the
```
Comment (mikewalch) [inline docs/applications.md]: Could be `Initialize your Fluo application`

File: docs/applications.md [hunk 6]
```diff
@@ -141,12 +126,123 @@ To create an observer, follow these steps:
 3.  Build a jar containing these classes and include this jar in the `lib/` directory of your Fluo
     application.
 4.  Configure your Fluo instance to use this observer provider by modifying the Application section of
-    [fluo-app.properties].
+    [fluo-app.properties]. Set `fluo.observer.provider` to the observer provider class name.
 5.  Initialize Fluo.  During initialization Fluo will obtain the observed columns from the 
     ObserverProvider and persist the columns in Zookeeper.  These columns persisted in Zookeeper
     are used by transactions to know when to trigger observers.
 6.  Start your Fluo instance so that your Fluo workers load the new observer.
```
Comment (mikewalch) [inline docs/applications.md]: Could be `Start your Fluo application so that`

File: docs/applications.md (hunk 2 above)
Comment (keith-turner) [inline docs/applications.md]: Based on this comment I removed some redundant info at the end of this section.  Is that what you were thinking of?

File: docs/applications.md (hunk 3 above)
Comment (keith-turner) [inline docs/applications.md]: I didn't understand why you wanted to make this change and didn't change anything.

File: docs/applications.md [hunk 7]
```diff
@@ -140,12 +135,123 @@ To create an observer, follow these steps:
...
+
+4. Run `fluo list` which connects to Fluo and lists applications to verify initialization.
+
+5. Run `fluo config myapp` to see what configuration is stored in ZooKeeper.
+
+## Starting your Fluo application
+
+Follow the instructions below to start a Fluo application which contains an oracle and multiple workers.
+
+1. Configure [fluo-env.sh] and [fluo-conn.properties] if you have not already.
```
Comment (mikewalch) [inline docs/applications.md]: No link at bottom of page for fluo-env.sh

File: docs/applications.md [hunk 8]
```diff
@@ -140,12 +135,123 @@ To create an observer, follow these steps:
...
+
+2. Edit your copy of [fluo-app.properties] and make sure to set the following:
+
+    * Class name of your ObserverProvider
+    * Paths to your Fluo observer jars
+    * Accumulo configuration
+    * DFS configuration
+
+   When configuring the observer section of fluo-app.properties, you can configure your instance for the
+   [phrasecount] application if you have not created your own application. See the [phrasecount]
```
Comment (mikewalch) [inline docs/applications.md]: No link for phrasecount

File: docs/applications.md [hunk 9]
```diff
@@ -17,8 +17,18 @@ limitations under the License.
 
 # Fluo Applications
 
-Once you have Fluo installed and running on your cluster, you can now run Fluo applications which
-consist of clients and observers.
+Once you have Fluo installed and running on your cluster, you can run Fluo applications consisting
+of clients and observers.  A client is a user started process that uses the Fluo API.  An Observer
+is user provided code run by Fluo Workers in response to notifications.  Notifications are set by
```
Comment (mikewalch) [inline docs/applications.md]: Could link to architecture.md here if users want to better understand components of an application.  This text could be incorporated into architecture.md to improve it.

**Pull request merged**